src/guides/getting-started.md

```diff
@@ -573,6 +573,58 @@ That's it! Let's try it out with `cargo run --bin publish_post 1`.
 Published post Diesel demo
 ```
 
+Additionally let's implement a possibility of fetching a single post. We will display the post id with its title.
+Notice `optional()` call. This returns `Option<Post>` instead of throwing an error, which we can then use in our matching pattern.
+
+::: code-block
+
+[src/bin/get_post.rs](https://github.com/diesel-rs/diesel/blob/2.1.x/examples/postgres/getting_started_step_3/src/bin/get_post.rs)
+
+```rust
+use self::models::Post;
+use diesel::prelude::*;
+use diesel_demo_step_3_sqlite::*;
+use std::env::args;
+
+fn main() {
+    use self::schema::posts::dsl::{id, posts};
+
+    let post_id = args()
+        .nth(1)
+        .expect("get_post requires a post id")
+        .parse::<i32>()
+        .expect("Invalid ID");
+    
+    let connection = &mut establish_connection();
+
+    let post = connection
+        .transaction(|connection| {
+            let post = posts.find(id).select(Post::as_select())
+                .first(connection)
+                .optional()?; // This allows for returning an Option<Post>, otherwise it will throw an error
+
+            match post {
+                Some(post) => Ok(post),
+                None => Err(diesel::result::Error::NotFound),
+            }
+        })
+        .unwrap_or_else(|_: diesel::result::Error| panic!("Unable to find post {}", post_id));
+
+    println!("Post with id: {} has a title: {}", post.id, post.title);
```

Comment:
I would rather like to do something like the following here:

```rust
match post {
     Some(post) => println!("Post with id: …"),
     None => println!("No post with id {} found")
}
```

Reply:
I had to add also a third case for error, but it looks a lot better this way